packages/playwright-test/src/reporters/azure.ts

```diff
@@ -0,0 +1,481 @@
+      };
+      if (!this.testApi)
+        this.testApi = await this.connection.getTestApi();
+      const runUpdatedResponse = await this.testApi.updateTestRun(runUpdate, this.projectName, this.runId as number);
+      this.log(colors.green(`Run ${this.runId} - ${runUpdatedResponse.state}`));
+    } catch (err) {
+      this.log(`Error on completing run ${err as string}`);
+    }
+  }
+
+  private log(message?: any) {
+    if (this.logging)
+      console.log(colors.magenta(`azure: ${message}`));
+
+  }
+
+  private getCaseIds(test: TestCase): string {
+    const regexp = /\[([\d,]+)\]/;
+    const results = regexp.exec(test.title);
+    if (results && results.length === 2)
+      return results[1];
+
+    return '';
+  }
+
+  private logTestItem(test: TestCase, testResult: TestResult) {
+    const map = {
+      failed: colors.red(`Test ${test.title} - ${testResult.status}`),
+      passed: colors.green(`Test ${test.title} - ${testResult.status}`),
+      skipped: colors.blue(`Test ${test.title} - ${testResult.status}`),
+      pending: colors.blue(`Test ${test.title} - ${testResult.status}`),
+      disabled: colors.gray(`Test ${test.title} - ${testResult.status}`),
+      timedOut: colors.gray(`Test ${test.title} - ${testResult.status}`)
+    };
+    if (testResult.status)
+      this.log(map[testResult.status]);
+
+  }
+
+  private async createRun(runName: string): Promise<TestInterfaces.TestRun> {
+    return new Promise<TestInterfaces.TestRun>(async (resolve, reject) => {
+      try {
+        this.checkProject(this.projectName).catch(e => {
+          reject(e);
+        });
+        const runModel: TestInterfaces.RunCreateModel = {
+          name: runName,
+          automated: true,
+          configurationIds: [1],
+          plan: { id: `${this.planId}` }
+        };
+        if (!this.testApi)
+          this.testApi = await this.connection.getTestApi();
+        const adTestRun = await this.testApi.createTestRun(runModel, this.projectName);
+        resolve(adTestRun);
+      } catch (e) {
+        reject(e);
+      }
+    }).catch(e => {
+      this.log(colors.red(`While creating test run.\n ${e}`));
+      this.isDisabled = true;
+      throw e;
+    });
+  }
+
+  private removePublished(testAlias: string): void {
+    const resultIndex = this.resultsToBePublished.indexOf(testAlias);
+    if (resultIndex !== -1)
+      this.resultsToBePublished.splice(resultIndex, 1);
+  }
+
+  private async checkProject(projectName: string): Promise<TeamProject | void> {
+    return new Promise<TeamProject>(async (resolve, reject) => {
+      try {
+        if (!this.coreApi)
+          this.coreApi = await this.connection.getCoreApi();
+        const project = await this.coreApi.getProject(projectName);
+        if (project)
+          resolve(project);
+        else
+          reject(`Project ${projectName} does not exist. Reporting is disabled.`);
+
+      } catch (e) {
+        reject(e);
+      }
+    }).catch(e => {
+      this.log(e);
```

Comment:
I don't like the general way of error handling, that it gets logged, then isDisabled is true and then after the actual function call you check if its disabled to return otherwise. Try to avoid using .catch, new Promise, either wrap the errors into new ones to make them readable, if some are expected or don't handle them with too much precaution.

Reply:
done